lib/proteus/kit.rb

```diff
@@ -1,32 +1,55 @@
 require "proteus/version"
+require "proteus/repos"
 require "thor"
 
 module Proteus
   class Kit < Thor
     include Thor::Actions
 
-    desc "url", "gets the git url"
-    def url(kit)
-      "https://github.com/thoughtbot/proteus-#{kit}.git"
+    no_commands do
+      def url(id)
+        kit = Proteus::REPOS[id.to_sym]
+        kit[:url]
+      end
+
+      def name(id)
+        kit = Proteus::REPOS[id.to_sym]
+        kit[:name]
+      end
+    end
+
+    desc "list", "shows a list of available kits"
+    def list
+      repos = Proteus::REPOS
+      repos.each do |id, repo|
+        puts "#{id} - #{repo[:name]}"
+      end
     end
 
     desc "new", "runs the command to clone a particular kit"
-    def new(kit_name, repo_name = nil)
-      repo_name ||= kit_name
-
-      if system "git ls-remote #{url(kit_name)} > /dev/null 2>&1"
-        puts "Starting a new proteus-#{kit_name} project in #{repo_name}"
-        system %{
-          git clone "#{url(kit_name)}" "#{repo_name}" &&
-          cd "#{repo_name}" &&
-          rm -rf .git &&
-          git init &&
-          git add . &&
-          git commit -am 'New proteus-#{kit_name} project' &&
-          cd -
-        }
+    def new(id, dir = id)
```

Comment:
This auto assignment is a little confusing here. I think I would assign `nil` to dir and then use `dir ||= id` inside the method for assignment or have a `dir` method that falls back to `id` if it's not assigned.